src/adapters/Cornerstone/Segmentation_4X.js

```diff
@@ -651,16 +688,36 @@ function checkSEGsOverlapping(
         );
 
         if (!alignedPixelDataI) {
-            console.warn(
-                "Individual SEG frames are out of plane with respect to the first SEG frame, this is not yet supported, skipping this frame."
-            );
+            // Individual SEG frames are out of plane with respect to the first SEG frame, this is not yet supported
+            i = i + numberOfFrames;
+            if (i >= groupsLen) {
+                i = i - numberOfFrames * numberOfSegs + 1;
+                if (i >= numberOfFrames) {
+                    break;
+                }
+            }
 
-            inPlane = false;
-            break;
+            continue;
         }
 
         const segmentIndex = getSegmentIndex(multiframe, i);
 
+        if (segmentIndex === undefined) {
+            console.warn(
+                "Could not retrieve the segment index, skipping this frame. "
```

Comment:
We should be thinking of ways to send this kind of information back to the app level in a consumable way.  End users shouldn't be needing to look for console.warn messages.

Reply:
I agree, we can talk about this at the next OIHF-IDC meeting. What about we expand the new Logger service implemented by @igoroctaviano to catch also info/log/warn and not only errors? https://github.com/OHIF/Viewers/pull/2283

Reply:
It's sticky to catch, identify and create messages for all possible errors coming from dcmjs / thrid party in OHIF. Maybe dcmjs needs to define a new error layer that will throw well-defined error messages + indexes?

Reply:
I agree with you @igoroctaviano , but I would do this in a separate PR.

Reply:
Yes, I agree.  We just did something similar for the save dialog in Slicer, where the messages from the dependency libraries are captured into message buffers that are exposed behind icons in the UI dialog.  Definitely worth its own discussion and PR.